src/palace/manager/util/http.py

```diff
@@ -99,55 +96,28 @@ def __init__(
         self,
         url_or_service: str,
         message: str,
+        response: Response,
         debug_message: str | None = None,
-        status_code: int | None = None,
     ):
         """Indicate that a remote integration has failed.
 
         `param url_or_service` The name of the service that failed
            (e.g. "Overdrive"), or the specific URL that had the problem.
         """
+        if debug_message is None:
+            debug_message = (
+                f"Status code: {response.status_code}\nContent: {response.text}"
+            )
+
         super().__init__(url_or_service, message, debug_message)
         # to be set to 500, etc.
-        self.status_code = status_code
-
-    @classmethod
-    def from_response(
-        cls, url: str, message: str, response: HttpResponseTuple | Response
-    ) -> Self:
-        """Helper method to turn a `requests` Response object into
-        a BadResponseException.
-        """
-        if isinstance(response, tuple):
-            # The response has been unrolled into a (status_code,
-            # headers, body) 3-tuple.
-            status_code, _, content_bytes = response
-            # The HTTP content response is a bytestring that we want to
-            # convert to unicode for the debug message.
-            if content_bytes:
-                content = content_bytes.decode("utf-8")
-            else:
-                content = ""
-        else:
-            status_code = response.status_code
-            content = response.text
-
-        return cls(
-            url,
-            message,
-            status_code=status_code,
-            debug_message="Status code: %s\nContent: %s"
-            % (
-                status_code,
-                content,
-            ),
-        )
+        self.response = response
```

Comment:
Minor: ^ Does the above comment make sense anymore?

Reply:
Good call, will 🪓